tasks/deploy-orochi-network.ts

```diff
@@ -6,17 +6,33 @@ import { env } from '../env';
 import { getAddress, isAddress, keccak256 } from 'ethers';
 import { HexString, OrandEncoding } from '@orochi-network/utilities';
 import { getWallet } from '../helpers/wallet';
+import EthJsonRpc from '../helpers/provider';
+
+export const CHAIN_NEED_CUSTOM_PROVIDER = [196, 7225878];
+export const GAS_LESS_BLOCK_CHAIN = [7225878];
+export const GAS_LIMIT_IN_GAS_LESS_BLOCKCHAIN = 11000000n;
 
 const OPERATORS = env.OROCHI_OPERATOR.split(',').map((op) => op.trim());
 
 task('deploy:orochi', 'Deploy Orochi Network contracts').setAction(
   async (_taskArgs: any, hre: HardhatRuntimeEnvironment) => {
+    if (!hre.network.config.chainId) {
+      throw new Error('Invalid chainId');
+    }
+    const needCustomProvider = CHAIN_NEED_CUSTOM_PROVIDER.includes(hre.network.config.chainId);
+    const isGasLessBlockchain = CHAIN_NEED_CUSTOM_PROVIDER.includes(hre.network.config.chainId);
     // Public key
     let pk = env.OROCHI_PUBLIC_KEY.replace(/^0x/gi, '').trim();
     let correspondingAddress = getAddress(`0x${keccak256(`0x${pk.substring(2, 130)}`).substring(26, 66)}`);
     // Get deployer account
-    const { chainId } = await hre.ethers.provider.getNetwork();
-    const account = await getWallet(hre, chainId);
+    const provider = needCustomProvider
```

Comment:
This code looks so ugly, we can move the logic into `getWallet()` method, it will be better and clean.

Reply:
Fixed